### ISSUE TITLE
Enable free-form input of math expressions

### ISSUE THREAD

**Issue by grochocki** [microsoft/calculator#620]
**Problem Statement**
Today, when you input an expression, you are committing previous input with each operator, which means that you cannot easily edit or undo what you have already committed. We also have a key mapping such that keyboard keys are mapped to specific button inputs. This means that to enter "cos()", you actually press o. This is unintuitive and makes entering equations difficult.

We also currently only support linear input, which means we display this:

x ^ 2 + √(2x)

instead of this:

![image](https://user-images.githubusercontent.com/1414423/62503564-92b1e980-b7a9-11e9-8f73-7933cf7ead89.png)

**Evidence or User Insights**
Our current keyboard mapping is unintuitive and requires extensive documentation to educate users (#157). Not only is free-form input more intuitive, there have been several issues opened that would require support for it as a pre-requisite (e.g., #163, #519, #150).

**Proposal**
Enable free-form input across calculator modes, which means that users can place the cursor anywhere inside of an equation input box to edit the expression. This also means that users can type or paste in "cos(x)", which feels natural, as this is how you would write this expression up on a whiteboard.

> Note: This is the same input behavior that we are going to have in graphing mode.

**Goals**
* Users can input their equations with free-form input
* Expressions are displayed in rich [MathML](https://www.w3.org/Math/) styling format

**Non-Goals**
* Build a parser (this is covered by #526, which is a prerequisite for this work)
* Introduce new input methods other than keyboard (e.g., inking input is covered by #24)

**Low-Fidelity Concept**

![image](https://user-images.githubusercontent.com/1414423/62503564-92b1e980-b7a9-11e9-8f73-7933cf7ead89.png)
<sub>Just an example of rendering as MathML</sub>

**Requested Assignment**
I'm just suggesting this idea.  I don't want to implement it.

**Comment by MicrosoftIssueBot** [comment]
This is your friendly Microsoft Issue Bot. I've seen this issue come in and have gone to tell a human about it.

**Comment by grochocki** [comment]
This work is definitely dependent on #526, and is likely something we would want to pursuit it parallel. This feature is also dependent on MathML rendering.

**Comment by SavoySchuler** [comment]
@adambarlow this issue shows the MathML rendering that you were proposing for [NumberBox](https://github.com/microsoft/microsoft-ui-xaml/issues/483).
